src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Action/VirtualMachineActionBaseCmdlet.cs

```diff
@@ -0,0 +1,61 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.Commands.Compute
+{
+    public abstract class VirtualMachineActionBaseCmdlet : VirtualMachineBaseCmdlet
+    {
+        protected const string ResourceGroupNameParameterSet = "ResourceGroupNameParameterSetName";
+        protected const string IdParameterSet = "IdParameterSetName";
+
+        [Parameter(
+           Mandatory = true,
+           Position = 0,
+           ParameterSetName = ResourceGroupNameParameterSet,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+           Mandatory = true,
+           Position = 0,
+           ParameterSetName = IdParameterSet,
```

Comment:
That is an interesting question.  Currently all child cmdlets of this base method do not have an additional parameter set. I think a child cmdlet can have more parameter sets in addition to these parameter sets.